docs/docs/sample-scripts/spo/copy-librarys-folders-and-contents-to-another-library/index.md

```diff
@@ -0,0 +1,68 @@
+---
+tags:
+  - files
+  - folders
+  - libraries
+  - migration
+---
+
+# Copy library's folders and contents to another library
+
+Author: [Nico De Cleyre](https://www.nicodecleyre.com)
+
+The following script shows how you can copy all library's folders and contents to another library
+
+=== "PowerShell"
+
+    ```powershell
+    param
+    (
+        [Parameter(Mandatory = $true, HelpMessage = "The source site url")][string] $SiteURL,
+        [Parameter(Mandatory = $true, HelpMessage = "The server- or site-relative URL of the source document library")][string] $SourceFolder,
+        [Parameter(Mandatory = $true, HelpMessage = "The server- or site-relative URL of the destination document library")][string] $DestinationFolder,
+        [Parameter(Mandatory = $false, HelpMessage = "The destination site url")][string] $DestinationUrl
+    )
+    try {
+        #Connect to m365
+        $M365Status = m365 status
+        if ($M365Status -match "Logged Out") {
+            Write-Host "Logging in the User!"
+            m365 login --authType browser
+        }
+
+        # Set the destination URL the same as the source URL if it's not given
+        if($DestinationUrl -eq $null){
+            $DestinationUrl = $SiteURL
+        }
+
+        $SourceFolders = m365 spo folder list --webUrl $SiteURL --parentFolderUrl $SourceFolder | ConvertFrom-Json
+        foreach($Folder in $SourceFolders){
+            if($Folder.Name -eq "Forms"){
+                continue
+            }
+            $DestinationFolderUrl = $Folder.ServerRelativeUrl.Replace($SourceFolder, $DestinationFolder)
+
+            # Create level 1 folders first
+            Write-Host "Creating '$($Folder.Name)' in '$DestinationFolder'" -ForegroundColor Yellow
+            $AddedFolder = m365 spo folder add --webUrl $DestinationUrl --parentFolderUrl $DestinationFolder --name $Folder.Name
```

Comment:
just wondering. If we do not do anything with the `$AddedFolder` do we need this variable? I think that if you do not want to display the output of the command in the log you could add ` | Out-null` at the end of the command 🤔